go/internal/feast/registry.go

```diff
@@ -228,89 +248,113 @@ func (r *Registry) listOnDemandFeatureViews(project string) []*OnDemandFeatureVi
+	_, err := r.getRegistryProto(allowCache)
+	if err != nil {
+		return nil, err
+	}
 	if featureViews, ok := r.cachedFeatureViews[project]; !ok {
-		return nil, errors.New(fmt.Sprintf("project %s not found in getFeatureView", project))
+		return nil, fmt.Errorf("project %s not found in getFeatureView", project)
 	} else {
 		if featureViewProto, ok := featureViews[featureViewName]; !ok {
-			return nil, errors.New(fmt.Sprintf("featureView %s not found inside project %s", featureViewName, project))
+			return nil, fmt.Errorf("featureView %s not found inside project %s", featureViewName, project)
 		} else {
 			return NewFeatureViewFromProto(featureViewProto), nil
 		}
 	}
 }
 
-func (r *Registry) getFeatureService(project, featureServiceName string) (*FeatureService, error) {
+func (r *Registry) getFeatureService(project, featureServiceName string, allowCache bool) (*FeatureService, error) {
+	_, err := r.getRegistryProto(allowCache)
+	if err != nil {
+		return nil, err
+	}
 	if featureServices, ok := r.cachedFeatureServices[project]; !ok {
-		return nil, errors.New(fmt.Sprintf("project %s not found in getFeatureService", project))
+		return nil, fmt.Errorf("project %s not found in getFeatureService", project)
 	} else {
 		if featureServiceProto, ok := featureServices[featureServiceName]; !ok {
-			return nil, errors.New(fmt.Sprintf("featureService %s not found inside project %s", featureServiceName, project))
+			return nil, fmt.Errorf("featureService %s not found inside project %s", featureServiceName, project)
 		} else {
 			return NewFeatureServiceFromProto(featureServiceProto), nil
 		}
 	}
 }
 
-func (r *Registry) getOnDemandFeatureView(project, onDemandFeatureViewName string) (*OnDemandFeatureView, error) {
+func (r *Registry) getOnDemandFeatureView(project, onDemandFeatureViewName string, allowCache bool) (*OnDemandFeatureView, error) {
+	_, err := r.getRegistryProto(allowCache)
+	if err != nil {
+		return nil, err
+	}
 	if onDemandFeatureViews, ok := r.cachedOnDemandFeatureViews[project]; !ok {
-		return nil, errors.New(fmt.Sprintf("project %s not found in getOnDemandFeatureView", project))
+		return nil, fmt.Errorf("project %s not found in getOnDemandFeatureView", project)
 	} else {
 		if onDemandFeatureViewProto, ok := onDemandFeatureViews[onDemandFeatureViewName]; !ok {
-			return nil, errors.New(fmt.Sprintf("onDemandFeatureView %s not found inside project %s", onDemandFeatureViewName, project))
+			return nil, fmt.Errorf("onDemandFeatureView %s not found inside project %s", onDemandFeatureViewName, project)
 		} else {
 			return NewOnDemandFeatureViewFromProto(onDemandFeatureViewProto), nil
 		}
 	}
 }
 
-func (r *Registry) getRequestFeatureView(project, requestFeatureViewName string) (*RequestFeatureView, error) {
+func (r *Registry) getRequestFeatureView(project, requestFeatureViewName string, allowCache bool) (*RequestFeatureView, error) {
```

Comment:
I really don't like this `allowCache` bool that we pass around, even in the python implementation. For the go server can we always read the in-memory version `cachedRegistry`, and just support a different couple of methods - one to start a goroutine that refreshes on an interval, and another to force a refresh?

Reply:
So what you're saying is just remove allowCache altogether?

Reply:
I'd want to yeah, I don't think the interface needs it

Reply:
Ok I removed the allowcache interface and added a go routine. We already have a force refresh function and now we always read from the in memory cache.